src/Mod/Draft/draftutils/grid_observer.py

```diff
@@ -0,0 +1,133 @@
+# ***************************************************************************
+# *   Copyright (c) 2009, 2010 Ken Cline <cline@frii.com>                   *
+# *   Copyright (c) 2023 FreeCAD Project Association                        *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+"""Provide the grid observer and background color change for the Draft Workbench.
+"""
+import lazy_loader.lazy_loader as lz
+
+import FreeCAD
+import FreeCADGui
+from draftutils import gui_utils
+from draftutils import utils
+
+
+# View observer code to update the Draft_ToggleGrid command button to reflect
+# the grid's visibility status
+# Based on view observer code to update the Draft Tray
+if FreeCAD.GuiUp:
+    import PySide.QtCore as QtCore
+    import PySide.QtGui as QtGui
+    from PySide import QtWidgets
+    from draftutils.todo import ToDo
+
+    def _update_gui():
+        try:
+            view = gui_utils.get_3d_view()
+            if view is None:
+                mw = FreeCADGui.getMainWindow()
+                if mw.findChild(QtWidgets.QToolBar, "Draft snap"):
+                    tbSnap = mw.findChild(QtWidgets.QToolBar, "Draft snap")
+                    btnGrid = _find_grid_toolbutton(
```

Comment:
"assignment-from-no-return" lint warnings need to be addressed.

Reply:
Fixed with last commit.